src/common/QskRgbValue.h

```diff
@@ -235,4 +235,335 @@ namespace QskRgb
+                }
+                if constexpr ( sizeof...( Cs ) == rrggbbaa )
+                {
+                    const auto r = ( rgba >> ( 6 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto a = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+            }
+
+            // converts a hex literal from '0x[AA]RRGGBB' to '0xAARRGGBB' integer
+            template< char... Cs >
+            constexpr QRgb operator""_argb()
+            {
+                constexpr auto rrggbb = 8;
+                constexpr auto aarrggbb = 10;
+                static_assert( sizeof...( Cs ) == rrggbb || sizeof...( Cs ) == aarrggbb,
+                    "invalid color literal length" );
+                constexpr auto rgba = parse_hex_literal_0< Cs... >();
+                if constexpr ( sizeof...( Cs ) == rrggbb )
+                {
+                    const auto r = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    const auto a = 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+                if constexpr ( sizeof...( Cs ) == aarrggbb )
+                {
+                    const auto r = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    const auto a = ( rgba >> ( 6 * 4 ) ) & 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+            }
+
+#define QSK_REQUIRE_CONSTEXPR_LITERAL 1
+#ifdef QSK_REQUIRE_CONSTEXPR_LITERAL
+            // RGBA hex string to QRgb with defaulted alpha = 0xFF
+            static_assert( "#123456"_rgba == 0xFF123456, "not constexpr" );
+            static_assert( "#123456"_argb == 0xFF123456, "not constexpr" );
+
+            // ARGB hex string to QRgb with defaulted alpha = 0xFF
+            static_assert( "#AA112233"_argb == 0xAA112233, "not constexpr" );
+            static_assert( "#112233AA"_rgba == 0xAA112233, "not constexpr" );
+
+            // RGBA hex literal to QRgb with defaulted alpha = 0xFF
+            static_assert( 0x112233_rgba == 0xFF112233, "" );
+            static_assert( 0xaabbcc_rgba == 0xFFAABBCC, "" );
+            static_assert( 0xAABBCC_rgba == 0xFFAABBCC, "" );
+            static_assert( 0x112233aa_rgba == 0xaa112233, "" );
+
+            // ARGB hex literal to QRgb with defaulted alpha = 0xFF
+            static_assert( 0x112233_argb == 0xFF112233, "" );
+            static_assert( 0xaabbcc_argb == 0xFFAABBCC, "" );
+            static_assert( 0xAABBCC_argb == 0xFFAABBCC, "" );
+            static_assert( 0x112233aa_argb == 0x112233aa, "" );
+#endif
+        }
+
+        namespace color
```

Comment:
@uwerat rename to `namespace qcolor` to be consistent with `namespace qrgb`?

Reply:
Please no qrgb and no qcolor - even for nested namespaces. 'Q' or 'q' names are reserved for Qt classes.

Not sure if the "literals" namespace is necessary as the definitions are already inside of QskRgb. I guess it would also be better to use different literals ( or function names ) instead of the additioal rgb/color namespace nesting. Otherwise users always have to qualify each literal when using QColor ad QRgb in the same file.